src/secret_storage.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper module used to manage interactions with Synapse secrets."""
+
+import logging
+import typing
+from secrets import token_hex
+
+import ops
+from ops.jujuversion import JujuVersion
+
+import actions
+import synapse
+
+PEER_RELATION_NAME = "synapse-peers"
+# Disabling it since these are not hardcoded password
+SECRET_ID = "secret-id"  # nosec
+SECRET_KEY = "secret-key"  # nosec
+
+logger = logging.getLogger(__name__)
+
+
+def _update_peer_data(charm: ops.CharmBase, container: ops.model.Container) -> None:
+    """Update peer data if needed.
```

Comment:
I feel like at least part of the comments in the method should be in the docstring as the docstring doesn't tell much about what the method does

Reply:
Good point, docstring expanded.